lib/rack/utf8_sanitizer.rb

```diff
@@ -8,6 +8,7 @@ module Rack
   class UTF8Sanitizer
     StringIO = ::StringIO
     NULL_BYTE_REGEX = /\x00/.freeze
+    NULL_BYTE_STRING_REGEX = Regexp.new('\\\u0000').freeze
```

Comment:
I'm quite confused about this PR. Isn't `\\\u0000` matching normal text that also happens to be a Ruby string escape sequence? There shouldn't be any reason to remove it from the input unless you're doing something rather strange with the string.

What problem does this actually solve?